sql: remove pre-23.1 version gates

RaduBerinde opened the pull request:
This PR is only for the top commit (it is on top of #114306).

This change removes pre-23.1 version gates in SQL code.
    
Informs: #112501
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114307)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
I have some nits, but 2nd commit <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 39 of 39 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-:-Nj4z5HK724M0O9mkesN:birl5sw)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andyyang890, @dt, @fqazi, @JeffSwenson, and @RaduBerinde)

___
*[`pkg/sql/crdb_internal.go` line 963 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4utHp-qlbdFqTJYrA:-Nj4utHp-qlbdFqTJYrB:beqwu5w) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/crdb_internal.go#L963)):*
> ```Go
> 
> func getInternalSystemJobsQueryFromClusterVersion(
> 	ctx context.Context, version clusterversion.Handle, predicate systemJobsPredicate,
> ```

nit: we could remove `version` argument.
___
*[`pkg/sql/crdb_internal.go` line 1119 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4v5InAShvBC8dR45U:-Nj4v5InAShvBC8dR45V:bqt9lh8) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/crdb_internal.go#L1119)):*
> ```Go
> )
> 
> func getCRDBInternalJobsTableTypeFilter(ctx context.Context, version clusterversion.Handle) string {
> ```

nit: this helper now seems redundant.
___
*[`pkg/sql/schema_changer.go` line 1371 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4vjF4CWcQD4ewrIJd:-Nj4vjF4CWcQD4ewrIJe:b-sw2jha) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/schema_changer.go#L1371)):*
> ```Go
> 	gcJobRecord := CreateGCJobRecord(
> 		jobDesc, sc.job.Payload().UsernameProto.Decode(), indexGCDetails,
> 		false, /* useLegacyGCJob */
> ```

nit: I think we could remove `useLegacyGCJob` argument as well as `storage.CanUseMVCCRangeTombstones`. Consider dropping a TODO somewhere.
___
*[`pkg/sql/show_stats.go` line 100 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4vzpY-JkA2GqShPhz:-Nj4vzpY-JkA2GqShPi-:b-l2gt2u) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/show_stats.go#L100)):*
> ```Go
> 		return nil, err
> 	}
> 	columns := showTableStatsColumnsPartialStatisticsVer
> ```

nit: we should remove old `showTableStatsColumns` and rename `showTableStatsColumnsPartialStatisticsVer` to `showTableStatsColumns`.
___
*[`pkg/sql/show_stats.go` line 115 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4wEQO4LGplibqEEo9:-Nj4wEQO4LGplibqEEoA:bmrx1y3) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/show_stats.go#L115)):*
> ```Go
> 			// TODO(yuzefovich): refactor the code to use the iterator API
> 			// (currently it is not possible due to a panic-catcher below).
> 			partialPredicateCol := `
> ```

nit: we could include these two columns into the query string right away and remove the need for `fmt.Sprintf`.
___
*[`pkg/sql/show_stats.go` line 405 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4wU5yDbwlla17yhiE:-Nj4wU5yDbwlla17yhiF:b-3d7qv8) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/show_stats.go#L405)):*
> ```Go
> 	}
> 
> 	row = append(row, partialPredicate)
> ```

nit: include this straight into the assignment above?
___
*[`pkg/sql/tenant_creation.go` line 312 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4wqouD14le3Vp7--O:-Nj4wqouD14le3Vp7--P:b-mc8cs8) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/tenant_creation.go#L312)):*
> ```Go
> 	switch info.DataState {
> 	case mtinfopb.DataStateReady:
> 		info.DeprecatedDataState = mtinfopb.ProtoInfo_READY
> ```

nit: consider dropping a TODO to remove this since we don't need to maintain compatibility with pre-v23.1.
___
*[`pkg/sql/catalog/lease/lease.go` line 1403 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4xLMS8AF9DMIm0yPl:-Nj4xLMS8AF9DMIm0yPm:b-4h3lj1) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/catalog/lease/lease.go#L1403)):*
> ```Go
> 		// Read orphaned leases.
> 		const (
> 			queryWithRegion = `
> ```

nit: could remove `queryWithoutRegion` and do `s/queryWithRegion/query/`.
___
*[`pkg/sql/sqlinstance/instancestorage/instancestorage.go` line 380 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4yZ7w-yeXxY0COyY8:-Nj4yZ7w-yeXxY0COyY9:b-llfm5x) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdf92626a841423e6a22fe204475235875883f61/pkg/sql/sqlinstance/instancestorage/instancestorage.go#L380)):*
> ```Go
> 	if !s.settings.Version.IsActive(ctx, clusterversion.TODO_Delete_V23_1_SystemRbrReadNew) {
> 		oldCache := func(ctx context.Context) (instanceCache, error) {
> 			return newRangeFeedCache(ctx, s.oldRowCodec, s.clock, s.f)
> ```

nit: probably could remove `oldRowCodec`.
___
*[`pkg/sql/stats/stats_cache.go` line 558 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4yuMI82xmirT0OZ1B:-Nj4yuMI82xmirT0OZ1C:b-45w7tq) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/stats/stats_cache.go#L558)):*
> ```Go
> 	// since we don't rely on a precise order of expectedTypes when we check them
> 	// below.
> 	expectedTypes = append(expectedTypes,
> ```

nit: we could push this into the definition above.
___
*[`pkg/sql/stats/stats_cache.go` line 771 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4z0s_ClsxPALdd_pz:-Nj4z0s_ClsxPALdd_q-:b-3on8uk) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/stats/stats_cache.go#L771)):*
> ```Go
> 	ctx context.Context, tableID descpb.ID, forecast bool,
> ) ([]*TableStatistic, error) {
> 	partialPredicateCol := `
> ```

nit: ditto for inlining into the query directly.


<!-- Sent from Reviewable.io -->

JeffSwenson commented on pkg/server/settingswatcher/version_guard_test.go:
```diff
@@ -109,9 +109,8 @@ func TestVersionGuard(t *testing.T) {
 			checkVersions: map[clusterversion.Key]bool{
-				initialVersion:                        true,
-				maxVersion:                            true,
-				clusterversion.TODO_Delete_V23_1Start: true,
```
nit: I think `clusterversion.TODO_Delete_V23_1Start` should be replaced with `startVersion`

JeffSwenson commented on pkg/sql/sqlliveness/slstorage/slstorage.go:
```diff
@@ -247,29 +247,13 @@ func (s *Storage) isAlive(
+	return s.newKeyCodec
 }
 
 func (s *Storage) versionGuard(
```
nit: The versionGuard helpers in slstorage, sqlinstance, and lease can be deleted. I see you added a TODO to the lease helper, but not the slstorage or sqlinstance helpers.

jeffswenson reviewed (approved):
LGTM

RaduBerinde reviewed (commented):
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-:-Nj8_oZKCkflNuz2Qaql:bz5mair)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andyyang890, @dt, @fqazi, and @yuzefovich)

___
*[`pkg/sql/crdb_internal.go` line 963 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4utHp-qlbdFqTJYrA:-Nj8aLIY8frwv4ibLNwf:b-pur7fb) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/crdb_internal.go#L963)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: we could remove `version` argument.
</blockquote></details>

Done, simplified the code as well.
___
*[`pkg/sql/schema_changer.go` line 1371 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4vjF4CWcQD4ewrIJd:-Nj8a_Zm2NWmXcCA_BXI:b-k4xu65) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/schema_changer.go#L1371)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: I think we could remove `useLegacyGCJob` argument as well as `storage.CanUseMVCCRangeTombstones`. Consider dropping a TODO somewhere.
</blockquote></details>

It's a lot of code to untangle, I added a TODO for `CreateGCJobRecord` for now
___
*[`pkg/sql/sqlinstance/instancestorage/instancestorage.go` line 380 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj4yZ7w-yeXxY0COyY8:-Nj8bvM97sAkHwrZ_cwV:bcsbf4v) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdf92626a841423e6a22fe204475235875883f61/pkg/sql/sqlinstance/instancestorage/instancestorage.go#L380)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: probably could remove `oldRowCodec`.
</blockquote></details>

Done, and renamed `newRowCodec` to `rowCodec`.
___
*[`pkg/sql/sqlliveness/slstorage/slstorage.go` line 253 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/114307#gh-1391396213:-Nj8eis26LvMZOOGP-hX:b-2y3s10) ([raw file](https://github.com/cockroachdb/cockroach/blob/626568e53e75f94ffabc91611d1d051ab2d2a4af/pkg/sql/sqlliveness/slstorage/slstorage.go#L253)):*
<details><summary><i>Previously, JeffSwenson (Jeff Swenson) wrote…</i></summary><blockquote>

nit: The versionGuard helpers in slstorage, sqlinstance, and lease can be deleted. I see you added a TODO to the lease helper, but not the slstorage or sqlinstance helpers.
</blockquote></details>

Done, added TODOs.


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
I added a commit that cleans up the version guard stuff, PTAL

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-:-Nj8qVbi2dw7Qs_0FY9E:byw0j57)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andyyang890, @dt, @fqazi, and @yuzefovich)



<!-- Sent from Reviewable.io -->

jeffswenson reviewed (commented):
>I added a commit that cleans up the version guard stuff, PTAL
LGTM 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-:-Nj8y91zAuifqzW1MsJ6:bmyw9a1)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andyyang890, @dt, @fqazi, and @yuzefovich)



<!-- Sent from Reviewable.io -->

andyyang890 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114307)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @dt, @fqazi, @RaduBerinde, and @yuzefovich)

___
*[`pkg/clusterversion/cockroach_versions.go` line 208 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj8yA1FAGPWOjeeQ2qh:-Nj8yA1G5wMNYd1GRAo-:b-xwcj8b) ([raw file](https://github.com/cockroachdb/cockroach/blob/930758da9b69fa88cc675999b3981b941bdd45b2/pkg/clusterversion/cockroach_versions.go#L208)):*
> ```Go
> 	// TODO_Delete_V23_1RoleMembersTableHasIDColumns is the version where the
> 	// role_members system table has columns for ids.
> 	TODO_Delete_V23_1RoleMembersTableHasIDColumns
> ```

Is there a reason that some of the version gates for non-permanent upgrades are kept? e.g. `TODO_Delete_V23_1RoleMembersTableHasIDColumns`


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114307)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andyyang890, @dt, @fqazi, and @yuzefovich)

___
*[`pkg/clusterversion/cockroach_versions.go` line 208 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-Nj8yA1FAGPWOjeeQ2qh:-Nj90d9D27UE4-X4Y4r4:bnql38e) ([raw file](https://github.com/cockroachdb/cockroach/blob/930758da9b69fa88cc675999b3981b941bdd45b2/pkg/clusterversion/cockroach_versions.go#L208)):*
<details><summary><i>Previously, andyyang890 (Andy Yang) wrote…</i></summary><blockquote>

Is there a reason that some of the version gates for non-permanent upgrades are kept? e.g. `TODO_Delete_V23_1RoleMembersTableHasIDColumns`
</blockquote></details>

This PR covers only `sql`, that is still used from `backupccl` (will be cleaned up in a separate PR soon).


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+

RaduBerinde commented:
bors r-

I will investigate the failure, I thought it was a flaky test but I want to make sure.

RaduBerinde reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114307)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andyyang890, @dt, @fqazi, and @yuzefovich)

___
*[`pkg/sql/sqlliveness/slstorage/sessionid.go` line 71 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-NjEoXBO7Z2DzN5XGi_H:-NjEoXBO7Z2DzN5XGi_I:bruynf1) ([raw file](https://github.com/cockroachdb/cockroach/blob/9addc7b5edaee60f9ede6d6a8cf353ba6f676e18/pkg/sql/sqlliveness/slstorage/sessionid.go#L71)):*
> ```Go
> 	b := session.UnsafeBytes()
> 	if len(b) == legacyLen {
> 		return nil, nil, errors.Newf("unexpected legacy SessionID format")
> ```

@JeffSwenson I am hitting a strange issue. I am seeing a failure of `TestExpiringSessionsAndClaimJobsDoesNotTouchTerminalJobs`. The failure is that [this check](https://github.com/cockroachdb/cockroach/blob/3b465209b1eca4d361e266ff0f8a8213113586ec/pkg/jobs/registry_external_test.go#L147) finds a `claim_session_id` instead of `NULL`, even after a long time. I effectively bisected on my changes and it came down to these lines. If I revert these lines, the test passes.

Can you think of an explanation? Would we ever write a SessionID in legacy format even when all versions are at or beyond 23.1?


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
*[`pkg/sql/sqlliveness/slstorage/sessionid.go` line 71 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-NjEoXBO7Z2DzN5XGi_H:-NjF9QpH5W5B8ZYFOikw:b-d3jwsz) ([raw file](https://github.com/cockroachdb/cockroach/blob/9addc7b5edaee60f9ede6d6a8cf353ba6f676e18/pkg/sql/sqlliveness/slstorage/sessionid.go#L71)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

@JeffSwenson I am hitting a strange issue. I am seeing a failure of `TestExpiringSessionsAndClaimJobsDoesNotTouchTerminalJobs`. The failure is that [this check](https://github.com/cockroachdb/cockroach/blob/3b465209b1eca4d361e266ff0f8a8213113586ec/pkg/jobs/registry_external_test.go#L147) finds a `claim_session_id` instead of `NULL`, even after a long time. I effectively bisected on my changes and it came down to these lines. If I revert these lines, the test passes.

Can you think of an explanation? Would we ever write a SessionID in legacy format even when all versions are at or beyond 23.1?
</blockquote></details>

This is the stack trace of where it happens. The id in question is coming from `fetchExpiredSessionIDs`.

```
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.UnsafeDecodeSessionID({0x14001856f50?, 0x14003aac078?})
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/sessionid.go:74 +0x250
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*rbrEncoder).encode(0x1400104b740, {0x14001856f50, 0x10})
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/key_encoder.go:41 +0x34
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteOrFetchSession.func1({0x10921ce68, 0x14006300ba0}, 0x140053621e0)
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:337 +0xc4
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0x140053621e0, {0x10921ce68, 0x14006300ba0}, 0x14006195af8)
  github.com/cockroachdb/cockroach/pkg/kv/txn.go:1012 +0x60
github.com/cockroachdb/cockroach/pkg/kv.runTxn({0x10921ce68, 0x14006300ba0}, 0x14006300ba0?, 0x0?)
  github.com/cockroachdb/cockroach/pkg/kv/db.go:1055 +0x40
github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl(0x10921ce30?, {0x10921ce68, 0x14006300ba0}, 0xc600a40?, 0x1?, 0x0?, 0x1088c5c20?)
  github.com/cockroachdb/cockroach/pkg/kv/db.go:1018 +0xa8
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).txn(...)
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:308
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteOrFetchSession(0x14003e74340, {0x10921ce30?, 0x14001a206e0?}, {0x1400185> 6f50, 0x10})
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:327 +0x100
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteOrFetchSessionSingleFlightLocked.func1({0x10921ce30?, 0x14001a206e0?})
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:284 +0x68
```


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
*[`pkg/sql/sqlliveness/slstorage/sessionid.go` line 71 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/114307#-NjEoXBO7Z2DzN5XGi_H:-NjFAKj77paxWa6uchtm:blhp9gl) ([raw file](https://github.com/cockroachdb/cockroach/blob/9addc7b5edaee60f9ede6d6a8cf353ba6f676e18/pkg/sql/sqlliveness/slstorage/sessionid.go#L71)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

This is the stack trace of where it happens. The id in question is coming from `fetchExpiredSessionIDs`.

```
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.UnsafeDecodeSessionID({0x14001856f50?, 0x14003aac078?})
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/sessionid.go:74 +0x250
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*rbrEncoder).encode(0x1400104b740, {0x14001856f50, 0x10})
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/key_encoder.go:41 +0x34
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteOrFetchSession.func1({0x10921ce68, 0x14006300ba0}, 0x140053621e0)
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:337 +0xc4
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0x140053621e0, {0x10921ce68, 0x14006300ba0}, 0x14006195af8)
  github.com/cockroachdb/cockroach/pkg/kv/txn.go:1012 +0x60
github.com/cockroachdb/cockroach/pkg/kv.runTxn({0x10921ce68, 0x14006300ba0}, 0x14006300ba0?, 0x0?)
  github.com/cockroachdb/cockroach/pkg/kv/db.go:1055 +0x40
github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl(0x10921ce30?, {0x10921ce68, 0x14006300ba0}, 0xc600a40?, 0x1?, 0x0?, 0x1088c5c20?)
  github.com/cockroachdb/cockroach/pkg/kv/db.go:1018 +0xa8
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).txn(...)
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:308
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteOrFetchSession(0x14003e74340, {0x10921ce30?, 0x14001a206e0?}, {0x1400185> 6f50, 0x10})
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:327 +0x100
github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteOrFetchSessionSingleFlightLocked.func1({0x10921ce30?, 0x14001a206e0?})
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:284 +0x68
```
</blockquote></details>

Ah, I feel silly now.. It's the test itself that's inserting these IDs:
```
terminalClaims[i] = uuid.MakeV4().GetBytes() // bogus claim
```


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
sql: remove pre-23.1 version gates

This change removes pre-23.1 version gates in SQL code and simplifies
the remaining code.

Informs: #112501
Release note: None

RaduBerinde commented:
bors r+

RaduBerinde commented:
Failure looks like an infra flake, trying again.

bors r+
